src/callbacks/callbacks.jl

```diff
@@ -283,7 +283,54 @@ function save_to_disk_func(integrator)
         precip_diagnostic = NamedTuple()
     end
 
-    if p.atmos.turbconv_model isa TC.EDMFModel
+    # Adds a prefix to the front of each name in the named tuple. This function
+    # is not type stable, but that probably doesn't matter for diagnostics.
+    add_prefix(diagnostics::NamedTuple{names}, prefix) where {names} =
+        NamedTuple{Symbol.(prefix, names)}(values(diagnostics))
+
+    cloud_fraction(ts, area::FT) where {FT} =
+        TD.has_condensate(thermo_params, ts) && area > 1e-3 ? FT(1) : FT(0)
+
+    if p.atmos.turbconv_model isa EDMFX
+        (; ᶜu⁰, ᶜts⁰) = p
+        (; ᶜsgs⁰, ᶜsgsᴰ, ᶜuᴰ, ᶜtsᴰ) = diagnostic_edmfx_quantities(Y, p, t)
+        (; a_min) = p.atmos.turbconv_model
+        ᶜareaᴰ = ᶜsgsᴰ.ρa ./ TD.air_density.(thermo_params, ᶜtsᴰ)
+        ᶜarea⁰ = ᶜsgs⁰.ρa ./ TD.air_density.(thermo_params, ᶜts⁰)
+        ᶜcloud_fractionᴰ = cloud_fraction.(ᶜtsᴰ, ᶜareaᴰ)
+        ᶜcloud_fraction⁰ = cloud_fraction.(ᶜts⁰, ᶜarea⁰)
+        draft_diagnostics = (;
+            common_diagnostics(ᶜuᴰ, ᶜtsᴰ)...,
+            area = ᶜareaᴰ,
+            cloud_fraction = ᶜcloud_fractionᴰ,
+        )
+        env_diagnostics = (;
+            common_diagnostics(ᶜu⁰, ᶜts⁰)...,
+            area = ᶜarea⁰,
+            cloud_fraction = ᶜcloud_fraction⁰,
+            tke = divide_by_ρa.(ᶜsgs⁰.ρatke, ᶜsgs⁰.ρa, 0, Y.c.ρ, a_min),
+        )
+        if p.atmos.precip_model isa Microphysics1Moment
+            ᶜq_raiᴰ =
```

Comment:
We currently only treat precipitation as a grid mean quantity. So we don't have the draft and environment rain and snow. 

Sorry, I guess that breaks the symmetry here. Would it be possible to opt out rain and snow from the SGS variables?

Reply:
Yes, it's definitely possible; I can just add an `if`-block to the definition of `times_a` that makes it ignore `ρq_rai` and `ρq_sno`, and then remove the corresponding draft and environment quantities computed in `save_to_disk_func`. Do we want to check with Tapio first, though?